tsa-core/src/main/kotlin/org/usvm/machine/interpreter/inputdict/Modification.kt

```diff
@@ -0,0 +1,62 @@
+package org.usvm.machine.interpreter.inputdict
+
+import org.usvm.UBoolExpr
+import org.usvm.machine.TvmContext
+import org.usvm.machine.splitHeadTail
+import kotlin.collections.plus
+
+/**
+ * Represent the single modification over a set of dictionary keys.
+ * Whenever you encounter `l: List<Modification>`, assume that `l.first()`
+ * is the last modification applied (and `l.last()` is the first modification applied).
+ */
+sealed interface Modification {
+    data class Remove(
+        val key: KeyType,
+    ) : Modification
+
+    data class Store(
+        val key: KeyType,
+        val guard: UBoolExpr,
+    ) : Modification
+}
+
+/** @return the condition that would be of a corresponding key in `getKeys` */
+fun List<Modification>.createKeyCondition(
+    ctx: TvmContext,
+    t: KExtended,
```

Comment:
Is it true that it is assumed that `t` should be in the root dictionary (in its `symbols`)? If so, you should add it as a comment. Or at least give `t` a better name (or both).